Source/Core/Core/Core.h

```diff
@@ -140,7 +143,11 @@ void RunOnCPUThread(std::function<void()> function, bool wait_for_completion);
 
 // for calling back into UI code without introducing a dependency on it in core
 using StateChangedCallbackFunc = std::function<void(Core::State)>;
-void SetOnStateChangedCallback(StateChangedCallbackFunc callback);
+// Returns a handle
+int AddOnStateChangedCallback(StateChangedCallbackFunc callback);
+// Also invalidates the handle
+bool RemoveOnStateChangedCallback(int& handle);
```

Comment:
Our coding conventions recommend passing a pointer rather than a reference if the function modifies the input.
This makes things more clear at the call site.

Reply:
Making a typedef for the handle would force me to include Core.h in FPSCounter.h, which I'd like to avoid (the reason why I made the handle an int in the first place).
About the reference/pointer, I've followed the same way Config::AddConfigChangedCallback() and ControllerInterface::RegisterDevicesChangedCallback() do, so it should be fine because the functions they are registered to also modify the input, if I understood what you mean correctly.

Reply:
The functions you mention don't modify their input.
See the "Functions" section. https://github.com/dolphin-emu/dolphin/blob/master/Contributing.md

Reply:
I thought you were talking about something else. Fixed.